### ISSUE TITLE
Setting "Keep Markdown Line Break Space" not respected by setting "Remove Trailing Whitespace"

### ISSUE THREAD

**Issue by cjprybol** [atom/whitespace#115]
OSX 10.11.4
Atom 1.7.3
Whitespace 0.32.2

When the setting "Keep Markdown Line Break Space" is enabled, the setting "Remove trailing whitespace" still chomps away the trailing whitespace. The file is recognized as markdown by Atom. Video link provided for example.

https://www.dropbox.com/s/y6f3fypafej9qpc/whitespace_example.mov?dl=0

**Comment by lee-dohm** [comment]
I cannot replicate this using Atom v1.9.0-dev-5a9bd28 on Mac OS X 10.11.4, which uses whitespace v0.32.2 also. Here is what I tried:

1. Launch Atom
1. Go into Settings View
1. Search for the whitespace package under the Packages tab
1. Click the Settings button
1. Ensure that both "Remove Trailing Whitespace" and "Keep Markdown Line Break Whitespace" are enabled (see below for screenshot)
1. Open the `README.md` file from this repo
1. Add five spaces to the end of a line that has text on it
1. Press <kbd>Enter</kbd> once to ensure that the cursor is not on the same line as the trailing whitespace
1. Save the file using <kbd>Cmd+S</kbd>

Expected and Actual: The trailing whitespace is not trimmed. (See screenshot below)

<img width="761" alt="screen shot 2016-05-01 at 9 47 59 pm" src="https://cloud.githubusercontent.com/assets/1038121/14947371/0d7cb3f2-0fe7-11e6-8dfc-09b149ae0baa.png">

<img width="1170" alt="screen_shot_2016-05-01_at_9_51_29_pm" src="https://cloud.githubusercontent.com/assets/1038121/14947368/082fce70-0fe7-11e6-984f-60ecfa37d4bc.png">

Additionally, I added a line to the tests for the Markdown functionality to explicitly set the "removeTrailingWhitespace" setting to `true`. The tests prior to this relied on the fact that "removeTrailingWhitespace" is the default. The tests still passed.

I see what the problem might be though. In your video, the file is identified as "Markdown". In mine, it is identified as GitHub Markdown (the Markdown grammar that is installed with Atom):

<img width="310" alt="screen shot 2016-05-01 at 9 56 14 pm" src="https://cloud.githubusercontent.com/assets/1038121/14947398/b5d83896-0fe7-11e6-876c-ee7e8c96a334.png">

The exception in the code for leaving trailing whitespace in Markdown files is designed to recognize the built-in version of Markdown. That is probably why you're seeing different behavior. 

Here's the line of code that decides whether or not to remove trailing whitespace based on whether the file is GitHub Markdown:

https://github.com/atom/whitespace/blob/master/lib/whitespace.coffee#L71

If you can tell me the scope name of the version of Markdown you're using, I might be able to add that package's scope name to the list.

**Comment by cjprybol** [comment]
Hi @lee-dohm, thanks for looking into this so quickly! You were right about the scope. I had installed and been using [this general markdown language](https://atom.io/packages/language-markdown) package in place of the built in github flavored package and scope.

I can't recall now why I switched, but I believe it had to do with the extra support of R markdown files.

The whitespace package wasn't recognizing that language scope as a markdown file. I disabled that markdown package, re-enabled the included language-gfm package, restarted atom, and the trailing white space functionality now works as expected!
